src/app/hooks/ens.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMemo, useCallback } from "react";
+import { useProvider } from "app/hooks";
+import { AvatarResolver } from "@ensdomains/ens-avatar";
+import { http } from "viem";
+import { mainnet } from "viem/chains";
+import { createEnsPublicClient } from "@ensdomains/ensjs";
+
+const client = createEnsPublicClient({
+  chain: mainnet,
+  transport: http(),
+});
+
+const useEns = () => {
+  const provider = useProvider();
+
+  const getEnsName = useCallback(
+    async (address: string) => {
+      const ensNameLS = localStorage.getItem(`ENS_${address}`);
```

Comment:
Add expiration logic for every storage usage. Cause at one day users can change their name or avatar